src/symbolic_memory.ml

```diff
@@ -38,27 +38,30 @@ module M = struct
     | _ -> Log.err {|Unsupported symbolic value reasoning over "%a"|} Expr.pp v
 
   let grow m delta =
-    let delta = Int32.to_int @@ i32 delta in
-    let old_size = m.size * page_size in
-    m.size <- max m.size ((old_size + delta) / page_size)
+    let old_size = Value.I32.mul m.size page_size in
+    let new_size = Value.I32.(div (add old_size delta) page_size) in
```

Comment:
I see the `max` is not here anymore. Could you explain me why?

Reply:
Oh actually, it's encoded through the `Triop (Ite` right ?

Reply:
Yes, because it's symbolic we might need to encode the greater size in a Ite expression. But I use `Value.Bool.select_expr` so as to not create concrete Ite expressions